core/src/main/java/hudson/model/listeners/SaveableListener.java

```diff
@@ -76,6 +81,10 @@ public void unregister() {
      * Fires the {@link #onChange} event.
      */
     public static void fireOnChange(Saveable o, XmlFile file) {
+        if (!Functions.isExtensionsAvailable() && !NOTIFY_DURING_STARTUP) {
```

Comment:
Is this more reliable than, say, waiting on a countdown latch to notify on the right initialization stage? Just an idea on how this might be invoked as soon as possible rather than waiting for the next time for those files to change.

Reply:
Not sure adding concurrency is a good idea in the startup that is already a lot too complex